core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -69,19 +72,39 @@ public RESTCatalog(Function<Map<String, String>, RESTClient> clientBuilder) {
   public RESTCatalog(
       SessionCatalog.SessionContext context,
       Function<Map<String, String>, RESTClient> clientBuilder) {
-    this.sessionCatalog = new RESTSessionCatalog(clientBuilder, null);
+    this(context, clientBuilder, null);
+  }
+
+  @VisibleForTesting
+  RESTCatalog(
+      SessionCatalog.SessionContext context,
+      Function<Map<String, String>, RESTClient> clientBuilder,
+      BiFunction<SessionCatalog.SessionContext, Map<String, String>, FileIO> ioBuilder) {
+    this.sessionCatalog = createSessionCatalog(clientBuilder, ioBuilder);
     this.delegate = sessionCatalog.asCatalog(context);
     this.nsDelegate = (SupportsNamespaces) delegate;
     this.context = context;
     this.viewSessionCatalog = sessionCatalog.asViewCatalog(context);
   }
 
+  @VisibleForTesting
+  RESTSessionCatalog createSessionCatalog(
```

Comment:
nit: How many places is this used? Do we want to decrease the number of costructors?